processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -161,7 +161,7 @@ public SearchQueryRunner(Segment segment)
 
             Map<String, DimensionSelector> dimSelectors = Maps.newHashMap();
             for (String dim : dimsToSearch) {
-              dimSelectors.put(dim, cursor.makeDimensionSelector(dim));
+              dimSelectors.put(dim, cursor.makeDimensionSelector(dim, null));
```

Comment:
I think it might make more sense to update Search and Select queries to support proper dimensionSpecs with extraction as opposed to just simple dimensions, in which case we would not need the single argument method.